File: h2go-chaincodes/models/gdo.go
package models

type GDO struct {
	DocType    string    `json:"docType"`
	GdoID      string    `json:"gdoId"`
	RequestID  string    `json:"requestId"`
	AssetType  AssetType `json:"assetType"`
	IssueDate  string    `json:"issueDate"`
	ExpiryDate string    `json:"expiryDate"`
	OwnerID    string    `json:"ownerId"`
	Status     GdoStatus `json:"status"`
}

File: frontend/src/app/types/gdo.ts
export enum GdOStatus {
  ACTIVE = "ACTIVE",
  EXPIRED = "EXPIRED",
  USED = "USED",
}

export interface GdO {
  gdoId: string;
  requestId: string;
  assetType: string;
  issueDate: string;
  expiryDate: string;
  ownerId: string;
  status: GdOStatus | string;
}
